test/test_gateway.py

```diff
@@ -0,0 +1,110 @@
+from ceph_iscsi_config import gateway
+import logging
+import mock
+import subprocess
+import unittest
+
+
+class CephiSCSIGatewayTest(unittest.TestCase):
+
+    def setUp(self):
+        self.error = subprocess.CalledProcessError(
+            returncode=2,
+            cmd=["a command"],
+            output=b'some output')
+        self.run_ceph_mock_return = []
+        self.settings = mock.MagicMock(),
+
+        self.logger = logging.getLogger('rbd-target-api')
+        self.gw = gateway.CephiSCSIGateway(
+            self.logger,
+            self.settings,
+            'thishost')
+
+    def mock_run_ceph_cmd(self, cmd, stderr=None, shell=True):
+        return self.run_ceph_mock_return.pop()
+
+    @mock.patch.object(gateway.subprocess, 'check_output')
+    def test__run_ceph_cmd(self, check_output):
```

Comment:
s/test__run_ceph_cmd/test_run_ceph_cmd/ ?

Reply:
The method being tested starts with an underscore `_run_ceph_cmd` . Hence the double underscore in the test name. I'm happy to change the test name though if you'd prefer `test_run_ceph_cmd`

Reply:
Yeah, please.

Reply:
At the same time could update the `README` to give some explaination and a example for the test ?

Reply:
Hi @lxbsz, thank you for the review. I can't think what would be relevant to put in the README since this PR is just adding some unit tests or are you suggesting I add doc strings to the tests?

Reply:
I am how to run the test. You need to update the exact command and what should others do to run the test.

Such as the `Curl Examples:`.

Reply:
I thought the README was aimed at users who want to install, configure and use ceph-iscsi. The `Curl Examples` in the README is aimed at users who want to interact with the api via curl. Running unit tests seems to me to only be of interest to developers. Having said that I could add something like 

```
diff --git a/README b/README
index fe32bd2..b0c7455 100644
--- a/README
+++ b/README
@@ -135,3 +135,9 @@ here:
 http://docs.ceph.com/docs/master/rbd/iscsi-target-cli/
 
 can be used to create a iscsi-gateway.cfg and create a target.
+
+## Running Unit Tests
+
+The unit tests for ceph-isci can be run with:
+
+```tox -e py3```
```

Is that the sort of thing you are looking for?